### PR TITLE
Add tests for underlined text rule

### PR DESCRIPTION
**Pull request by pattonwebz**
I added some tests for this rule which didn't yet have any. 

It highlighted some edge cases that wouldn't be caught by the existing checking. Elements inside other elements that are underlined as well as not actually checking all underline capable css. The check was updated to handle these cases.

## Copilot Summary

This pull request enhances the accessibility rule for detecting inappropriate underlined text by improving the logic, updating checks, and adding comprehensive test coverage. Key changes include refining the evaluation logic for checks, simplifying the rule configuration, and introducing detailed Jest tests for validation.

### Updates to checks and rule logic:

* **`element-is-u-tag.js`**: Improved the `evaluate` function to handle cases where `tagName` might be undefined, ensuring robustness.
* **`element-with-underline.js`**: Enhanced the `evaluate` function to check both `text-decoration` and `text-decoration-line` styles for underline detection, improving accuracy.
* **`underlined-text.js`**: Simplified the rule's `selector` to exclude elements with links or link roles and updated the `none` checks to use the renamed `element_with_underline` check. Removed redundant logic for parent traversal.

### Comprehensive test coverage:

* **`underlinedText.test.js`**: Added extensive Jest tests to validate the updated rule and checks. The tests include a variety of passing and failing cases, covering static, dynamic, and edge-case scenarios for underlined text.

## Checklist

- [ ] PR is linked to the main issue in the repo
- [x] Tests are added that cover changes

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of elements without a tag name to prevent potential errors.
  * Enhanced detection of underlined elements by checking multiple CSS properties.

* **Refactor**
  * Simplified and updated the logic for identifying underlined text, using more precise selectors and conditions.

* **Tests**
  * Added comprehensive tests to verify correct detection of underlined text and ensure rule accuracy in various scenarios.

### CODE REVIEW

File: tests/jest/rules/underlinedText.test.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
+import axe from 'axe-core';
+
+beforeAll( async () => {
+	// Dynamically import the custom rule
+	const underlinedTextRuleModule = await import( '../../../src/pageScanner/rules/underlined-text.js' );
+	const elementWithUnderlineCheckModule = await import( '../../../src/pageScanner/checks/element-with-underline.js' );
```
[the remark this answers is not in the document]
Comment (pattonwebz) [reply tests/jest/rules/underlinedText.test.js:6]: Noted, but not fixing at this point in this PR. That's a bigger change across all rules and checks to decide on.
